tests/poll_stale_conversions_test.php

```diff
@@ -0,0 +1,194 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Unit test for local_smartmedia conversion class.
+ *
+ * @package    local_smartmedia
+ * @copyright  2019 Matt Porritt <mattp@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+require_once($CFG->dirroot . '/local/aws/sdk/aws-autoloader.php');
+
+use local_smartmedia\conversion;
+use local_smartmedia\aws_api;
+use local_smartmedia\aws_elastic_transcoder;
+use local_smartmedia\task\poll_stale_conversions;
+use Aws\CommandInterface;
+use Aws\Result;
+use Aws\MockHandler;
+use Aws\S3\Exception\S3Exception;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Unit test for local_smartmedia conversion class.
+ *
+ * @package    local_smartmedia
+ * @copyright  2019 Matt Porritt <mattp@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @group      local_smartmedia
+ */
+class local_smartmedia_poll_stale_conversions_testcase extends advanced_testcase {
```

Comment:
Should `execute()` also be covered by tests?

Reply:
Its small enough that I dont care. Just wiring bits together. I will test it functionally